src/ephemeral.rs

```diff
@@ -448,6 +451,11 @@ pub(crate) async fn delete_expired_messages(context: &Context, now: i64) -> Resu
                         (DC_CHAT_ID_TRASH, msg_id),
                     )?;
 
+                    transaction.execute(
+                        "DELETE FROM locations WHERE independent=1 AND id=?",
+                        (location_id,),
+                    )?;
```

Comment:
maybe a little optimisation wrt calling into SQL only if needed makes sense here:

```suggestion
                    if location_id != 0 {
                        transaction.execute(
                            "DELETE FROM locations WHERE independent=1 AND id=?",
                            (location_id,),
                        )?;
                    }
```